src/pkg/services/m365/api/graph/errors.go

```diff
@@ -133,16 +133,38 @@ func stackWithCoreErr(ctx context.Context, err error, traceDepth int) error {
 		err = clues.Stack(core.ErrAlreadyExists, err)
 	case isErrNotFound(ode, err):
 		err = clues.Stack(core.ErrNotFound, err)
+	default:
+		err = toErrByRespCode(ode, err)
```

Comment:
I like the logical flow of this method -- attempt to match to well-known error code (highly specific but brittle) and if no match, fall back to error code matching (less specific understanding of the error but still gives us a general sense of the root cause)